mods/tnt/init.lua

```diff
@@ -1,11 +1,6 @@
 tnt = {}
--- Default to enabled in singleplayer and disabled in multiplayer
-local singleplayer = minetest.is_singleplayer()
-local setting = minetest.setting_getbool("enable_tnt")
-if (not singleplayer and setting ~= true) or
-		(singleplayer and setting == false) then
-	return
-end
+-- Default to enabled when in singleplayer
+local enable_tnt = minetest.setting_getbool("enable_tnt") or minetest.is_singleplayer()
```

Comment:
You actually need to move that out into an extra if-clause checking for `nil` (see my other comment).
Otherwise you won't be able to disable tnt in singleplayer since it also falls back for **`false`**.

`or`-ing for fallback doesn't work well with boolean settings (but is great for anything else)